From 86b8e3955312b311cd5923751d48f198e3bcb736 Mon Sep 17 00:00:00 2001
From: Dinesh Karthikeyan <quic_dinek@quicinc.com>
Date: Wed, 23 Mar 2022 23:03:49 +0530
Subject: [PATCH] ath12k: Enable extended tx stats

To dump ext_tx_stats incomplete wbm_tx_stats hinders the read operation.
Remove the unfinished wbm_tx_stats code to enable ext_tx_stats
support in ath12k.

Note: wbm_tx_stats can be added later as separate feature.

Signed-off-by: Dinesh Karthikeyan <quic_dinek@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs_sta.c | 24 ++----------------------
 1 file changed, 2 insertions(+), 22 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_sta.c b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
index c8aebc3..926d94c 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_sta.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
@@ -290,18 +290,8 @@ static ssize_t ath12k_dbg_sta_dump_tx_stats(struct file *file,
 	const int size = 2 * 4096;
 	char *buf, mu_group_id[MAX_MU_GROUP_LENGTH] = {0};
 	u32 index;
-	static const char *fields[HAL_WBM_REL_HTT_TX_COMP_STATUS_MAX] = {
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_OK] = "Acked pkt count",
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_TTL] = "Status ttl pkt count",
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_DROP] = "Dropped pkt count",
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_REINJ] = "Reinj pkt count",
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_INSPECT] = "Inspect pkt count",
-			 [HAL_WBM_REL_HTT_TX_COMP_STATUS_MEC_NOTIFY] =
-								"MEC notify pkt count"
-			};
-	int idx;
-
-	if (!arsta->tx_stats || !arsta->wbm_tx_stats)
+
+	if (!arsta->tx_stats)
 		return -ENOENT;
 
 	buf = kzalloc(size, GFP_KERNEL);
@@ -435,16 +425,6 @@ static ssize_t ath12k_dbg_sta_dump_tx_stats(struct file *file,
 	len += scnprintf(buf + len, size - len,
 			 "ack fails\n %llu\n\n", arsta->tx_stats->ack_fails);
 
-	len += scnprintf(buf + len, size - len,
-			 "WBM tx completion stats of data pkts :\n");
-
-	for (idx = 0; idx <= HAL_WBM_REL_HTT_TX_COMP_STATUS_MEC_NOTIFY; idx++) {
-		len += scnprintf(buf + len, size - len,
-				 "%-23s :  %llu\n",
-				 fields[idx],
-				 arsta->wbm_tx_stats->wbm_tx_comp_stats[idx]);
-	}
-
 	spin_unlock_bh(&ar->data_lock);
 
 	if (len > size)
-- 
2.7.4

